include/boost/spirit/home/x3/binary/binary.hpp

```diff
@@ -0,0 +1,209 @@
+      , BOOST_SCOPED_ENUM(boost::endian::order) endian, std::size_t bits>
+    struct binary_lit_parser
+      : parser<binary_lit_parser<V, T, endian, bits> >
+    {
+        static bool const has_attribute = false;
+
+        binary_lit_parser(V n_)
+          : n(n_) {}
+
+        template <typename Iterator, typename Context, typename Attribute>
+        bool parse(Iterator& first, Iterator const& last
+          , Context& context, unused_type, Attribute& attr_param) const
+        {
+            x3::skip_over(first, last, context);
+
+            // Even if the endian types are not pod's (at least not in the
+            // definition of C++03) it seems to be safe to assume they are
+            // (but in C++0x the endian types _are_ PODs).
+            // This allows us to treat them as a sequence of consecutive bytes.
+            boost::endian::endian_arithmetic<endian, T, bits> attr_;
+
+#if defined(BOOST_MSVC)
+// warning C4244: 'argument' : conversion from 'const int' to 'foo', possible loss of data
+#pragma warning(push)
+#pragma warning(disable: 4244)
+#endif
+            attr_ = n;
+#if defined(BOOST_MSVC)
+#pragma warning(pop)
+#endif
+
+            unsigned char* bytes = reinterpret_cast<unsigned char*>(&attr_);
+
+            Iterator it = first;
+            for (unsigned int i = 0; i < sizeof(attr_); ++i)
+            {
+                if (it == last || *bytes++ != static_cast<unsigned char>(*it++))
+                    return false;
+            }
+
+            first = it;
+            x3::traits::move_to(attr_, attr_param);
+            return true;
+        }
+
+        V n;
+    };
+
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename T, BOOST_SCOPED_ENUM(boost::endian::order) endian, std::size_t bits>
+    struct any_binary_parser : parser<any_binary_parser<T, endian, bits > >
+    {
+
+        typedef T attribute_type;
+        static bool const has_attribute =
+            !is_same<unused_type, attribute_type>::value;
+
+        template <typename Iterator, typename Context, typename Attribute>
+        bool parse(Iterator& first, Iterator const& last
+          , Context& context, unused_type, Attribute& attr_param) const
+        {
+            x3::skip_over(first, last, context);
+
+            attribute_type attr_;
+            unsigned char* bytes = reinterpret_cast<unsigned char*>(&attr_);
+
+            Iterator it = first;
+            for (unsigned int i = 0; i < sizeof(attr_); ++i)
+            {
+                std::cout <<"Hey"<<std::endl;
+                if (it == last)
+                    return false;
+                *bytes++ = *it++;
+            }
+
+            first = it;
+            x3::traits::move_to(endian::conditional_reverse< endian, endian::order::native>(attr_), attr_param);
+            return true;
+        }
+
+        template <typename V>
+        binary_lit_parser< V, T, endian, bits> operator()(V n) const
```

Comment:
This might undergo some changes still once I get 'lazy' parsers back in. But you can ignore my comment form now if that does not seem make any sense.